### ISSUE TITLE
httpOnly?

### ISSUE THREAD

**Issue by Mbcomm** [pfpayments/prestashop-1.7#19]
We have been trying to get the new PF module to work with Prestashop since December.

Status remains at “PostFinance Checkout Processin” despite successful payment. The customer's shopping cart does not empty.

PostFinance support writes:

_Basically, this is the setting for our communication via the webbooks
path=/; domain=www.ballonshop.ch; secure; HttpOnly
is not allowed.
May I ask you to deactivate the “HttpOnly” setting via your web host._

**Disable HttpOnly? Really? On a webshop?**
 
Yesterday support wrote:
_I asked our technical partner again.
they confirm to me that the "httpOnly" function is not supported by PostFinance Checkout's Prestashop plugin.
This would have to be included as a future proposed change._

I can't believe the module doesn't accept httpOnly.

**Comment by andrewrowanwallee** [comment]
@Mbcomm could you provide us with the space id of the affected merchant so we can verify the webhook functionality?

**Comment by Mbcomm** [comment]
the space id is 38066

**Comment by andrewrowanwallee** [comment]
@Mbcomm is this a sporadic issue or is this a problem all of the time? I can see quite a lot of successful webhook invocations (containing the httponly flag) from within the portal. Thanks

**Comment by Mbcomm** [comment]
Yes, the call is successful after the shop owner manually sets the order status to “payment successful” in Prestashop. See example:
Order Feb 14 at 10:37 p.m
11 failed calls.
Product information such as delivery date will be lost in Prestashop.

The order is NOT recorded as “payment successful” despite confirmation of successful payment in Prestashop. Status is ALWAYS "processing" or "waiting".

<img width="686" alt="Bildschirmfoto 2024-02-15 um 17 09 06" src="https://github.com/pfpayments/prestashop-1.7/assets/159792528/53042752-fb76-46c1-a490-a6ea393933a3">

It's been an impossible situation since December. A lot of lost sales and a lot of effort.
PostFinance Support is useless. They installed and set up the module in PS. They have all the access, SSH, FTP, shop admin. But nothing works for weeks.

**Comment by andrewrowanwallee** [comment]
@Mbcomm so is this a sporadic issue or does this happen with every single transaction?

**Comment by andrewrowanwallee** [comment]
@Mbcomm thanks for clarifying. We're still investigating

**Comment by andrewrowanwallee** [comment]
@Mbcomm is this happening for all payment methods or only for specific payment methods e.,g. Twint?

**Comment by Mbcomm** [comment]
as i can see, this affects all payments thru PostFinance Module

**Comment by thibault-mambour** [comment]
hi @Mbcomm do you think you could share some access to your shop for us to run some test and reproduce the error or at least provide us some steps we could use to reproduce it ?
We are having difficulties reproducing locally on our own setup... 
as it is sensitive data, you can share it by email to thibault.mambour@wallee.com

**Comment by thibault-mambour** [comment]
hi @Mbcomm , we looked on our end and we do not think it is coming from our end as we don’t set any cookies that either aren’t httpOnly already or are set through the shop framework (thus outside of our direct control). Do you use any additional 3rd party plugins or custom setup ?

**Comment by Mbcomm** [comment]
hi thanks for support
My customer has been running the site successfully for years using the Postfinance payment module. With the new checkout module everything is now destroyed because the orders don't arrive correctly.
Postfinance Support has installed the new module in the shop and has not been able to get it to work correctly for two months. They insisted for 2 months that it was a so called "HttpOnly" problem with the server, which can't be the case. It's despairing.

Yes, of course it has numerous plugins like almost every shop, but no custom setups.
